core/modules/layout/includes/layout.class.inc

```diff
@@ -691,11 +691,26 @@ class Layout {
             $new_positions[$new_region_name][] = $uuid;
           }
           else {
-            unset($this->content[$uuid]);
+            $orphaned_blocks[] = $uuid;
           }
         }
       }
     }
+    if (!empty($orphaned_blocks)) {
+      // Assign orphan blocks to the default or last region,
+      // if the former doesn't exist.
+      end($new_positions);
+      $default_region = isset($new_layout_info['default region']) ? $new_layout_info['default region'] : key($new_positions);
+      $orphaned_blocks_titles = array();
+      foreach ($orphaned_blocks as $uuid) {
+        $new_positions[$default_region][] = $uuid;
+        $orphaned_blocks_titles[] = $this->content[$uuid]->getAdminTitle();
+      }
+      $message = t('The following blocks have been reassigned to the %new_region region. The regions they were placed in previously do not exist in the new layout template. You can move them to more appropriate regions:', array(
```

Comment:
This is much better now @herbdool 👍 ...I still think that the semicolon and the block list should come directly after the `The following blocks have been reassigned to the %new_region region` phrase, with the rest of the sentence. The `The regions they were placed in previously do not exist in the new layout template.` should be starting with a comma and a `because`, to tie it to the previous bit. The `You can move them to more appropriate regions.` bit should go after the list of blocks.

In any way, deferring to @jenlampton who usually has better words for things 😉

Reply:
I tried to make life easier for translators by not breaking it up into fragments. I took a look at another similar case in core. I can't recall what it was now but was quite similar.